### PR TITLE
Feature/condo/doma 2676/refaktoring for classifier

### PR DESCRIPTION
**Pull request by VKislov**
Partial replacement of using a three-level classifier with just using `classifierRule`. Correction of table column rendering functions related to this change on the ticket page.
Remove obsolete `TicketClassifier`. Rebase and resolve conflicts. Created migration and generated types.

### CODE REVIEW

**Comment by pahaz** [comment]
It look like there is no test.js changes. Could you please add this fields to the tests.

File: apps/condo/migrations/20220705094338-0143_remove_ticket_categoryclassifier_and_more.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+--
+-- Remove field categoryClassifier from ticket
+--
+SET CONSTRAINTS "Ticket_categoryClassifier_e01aafa7_fk_TicketCat" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_categoryClassifier_e01aafa7_fk_TicketCat";
+ALTER TABLE "Ticket" DROP COLUMN "categoryClassifier" CASCADE;
+--
+-- Remove field placeClassifier from ticket
+--
+SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id";
+ALTER TABLE "Ticket" DROP COLUMN "placeClassifier" CASCADE;
```
Comment (AntonAL) [inline apps/condo/migrations/20220705094338-0143_remove_ticket_categoryclassifier_and_more.js:16]: With this migration you will just wipe out all information on classifier. Data transfer SQL-queries should be executed first and then you can drop all unnecessary columns.

**Review by AntonAL — CHANGES_REQUESTED** [review]

**Comment by VKislov** [comment]
There were about 50 tickets with unidentified classifierRule. Because of this, such tickets will not be classified. It is necessary to write a script that put down the classifierRule everywhere in accordance with the combination of category, place, problem classifiers specified in the ticket.
Screenshot from 2022-07-11 10-50-07

**Review by dkoviazin — APPROVED** [review]

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 2]
```diff
@@ -190,23 +190,18 @@ const useChangedFieldMessagesOf = (ticketChange) => {
 
                 return !isEmpty(addressChangePostfix) ? `${value}${addressChangePostfix}` : value
             },
-            placeClassifierDisplayName: (field, value, type: TicketChangeFieldMessageType) => {
+            classifierRuleDisplayName: (field, value, type: TicketChangeFieldMessageType) => {
                 let placeClassifierToDisplay
-                let categoryClassifierToDisplay
-                let problemClassifierToDisplay
-
+                console.log(ticketChange['classifierRuleDisplayNameFrom'])
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:195]: ?

File: apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx [hunk 3]
```diff
@@ -116,7 +116,7 @@ const useChangedFieldMessagesOf = (ticketChange) => {
         ['assigneeDisplayName', AssigneeMessage],
         ['executorDisplayName', ExecutorMessage, { add: 'pages.condo.ticket.TicketChanges.executor.add', remove: 'pages.condo.ticket.TicketChanges.executor.remove' }],
         ['classifierDisplayName', ClassifierMessage],
-        ['placeClassifierDisplayName', ClassifierMessage],
+        ['classifierRuleDisplayName', ClassifierMessage],
```
Comment (pahaz) [inline apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:119]: classifierDisplayName

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+exports.up = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Remove field placeClassifier from ticket
+--
+SET CONSTRAINTS "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_placeClassifier_01e3ec3a_fk_TicketPlaceClassifier_id";
+ALTER TABLE "Ticket" DROP COLUMN "placeClassifier" CASCADE;
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:11]: Comment

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+--
+-- Remove field problemClassifier from ticket
+--
+SET CONSTRAINTS "Ticket_problemClassifier_f1bfcd39_fk_TicketProblemClassifier_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_problemClassifier_f1bfcd39_fk_TicketProblemClassifier_id";
+ALTER TABLE "Ticket" DROP COLUMN "problemClassifier" CASCADE;
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:16]: Comment

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+--
+-- Remove field categoryClassifierDisplayNameFrom from ticketchange
+--
+ALTER TABLE "TicketChange" DROP COLUMN "categoryClassifierDisplayNameFrom" CASCADE;
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:20]: Comment

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+--
+-- Remove field categoryClassifierDisplayNameTo from ticketchange
+--
+ALTER TABLE "TicketChange" DROP COLUMN "categoryClassifierDisplayNameTo" CASCADE;
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:24]: Comment

File: apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+--
+-- Remove field categoryClassifierIdFrom from ticketchange
+--
+ALTER TABLE "TicketChange" DROP COLUMN "categoryClassifierIdFrom" CASCADE;
```
Comment (pahaz) [inline apps/condo/migrations/20220721202935-0153_remove_ticket_placeclassifier_and_more.js:28]: Comment

**Pull request merged**